narwhals/_pandas_like/utils.py

```diff
@@ -294,7 +294,7 @@ def native_to_narwhals_dtype(native_column: Any, dtypes: DTypes) -> DType:
             native_column.dtype.pyarrow_dtype.list_size,
         )
     if dtype.startswith("struct"):
-        return dtypes.Struct()
+        return arrow_native_to_narwhals_dtype(native_column.dtype.pyarrow_dtype, dtypes)
```

Comment:
this gets me thinking, can the `if dtype.startswith(("large_list", "list")):` branch above be simplified to also just return this? (for a separate pr)

Reply:
I'll do that.